Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentTransaction.hs

```diff
@@ -47,5 +49,16 @@ findByTxnUUID :: Transactionable m => Text -> m (Maybe PaymentTransaction)
 findByTxnUUID txnUUID =
   findOne $ do
     transaction <- from $ table @PaymentTransactionT
-    where_ $ transaction ^. PaymentTransactionTxnUUID ==. val txnUUID
+    where_ $ transaction ^. PaymentTransactionTxnUUID ==. val (Just txnUUID)
+    return transaction
+
+findNewTransactionByOrderId :: Transactionable m => Id PaymentOrder -> m (Maybe PaymentTransaction)
+findNewTransactionByOrderId orderId =
+  findOne $ do
+    transaction <- from $ table @PaymentTransactionT
+    where_ $
+      Esq.isNothing (transaction ^. PaymentTransactionTxnUUID)
+        &&. transaction ^. PaymentTransactionOrderId ==. val (toKey orderId)
+
+    limit 1
```

Comment:
if one order can have more than 1 transactions then would we need order by created_at also right ?

Reply:
Fixed. Anyway second transaction with the same `order_id` and `txn_uuid=null` shouldn't be created